lib/jnpr/junos/console.py

```diff
@@ -274,13 +277,21 @@ def _tty_login(self):
             tty_args['console_has_banner'] = self.console_has_banner
             self.console = ('telnet', self._hostname, self.port)
             self._tty = Telnet(**tty_args)
+        elif self._mode.upper() == 'SSH':
+            tty_args['s_user'] = self.s_user
+            tty_args['s_passwd'] = self.s_passwd
+            tty_args['host'] = self._hostname
+            tty_args['port'] = self._port
+            tty_args['console_has_banner'] = self.console_has_banner
+            self.console = ('ssh', self._hostname, self.port)
+            self._tty = SSH(**tty_args)
         elif self._mode.upper() == 'SERIAL':
             tty_args['port'] = self._port
             self.console = ('serial', self._port)
             self._tty = Serial(**tty_args)
         else:
-            logger.error('Mode should be either telnet or serial')
-            raise AttributeError('Mode to be telnet/serial')
+            logger.error('Mode should be one of: telnet, serial, ssh')
```

Comment:
default connection is ssh only (netconf over ssh). We cant use `ssh` for console server connection trigger point.

Reply:
I have renamed to cs_ssh (console server ssh).